src/compas/geometry/primitives/circle.py

```diff
@@ -141,6 +141,14 @@ def __setitem__(self, key, value):
     def __iter__(self):
         return iter([self.plane, self.radius])
 
+    def __eq__(self, other):
+        try:
+            other_plane = other[0]
```

Comment:
Could it be that other is an `iterable` and index accessors cannot be used?

The case that comes to mind is that of built-in equivalences of primitives; in principle, an iterable could also be a circle, but I am not sure if we care for iters when it comes to these things. If not, then ignore this point.

Reply:
I lean toward the side of "if `other` doesn't allow index accessing, then it doesn't quack sufficiently  like a `Circle` or a list". 🦆

Reply:
fair enough! I just double checked the docs, and I believe there's nothing hinting at the possibility of using iterables, so I guess it's fine.

Reply:
```Frame.worldXY() == Polyline([[0,0,0], [1,0,0], [0,1,0]])``` returns `True`. so that's kind of funny.

Reply:
🤔 maybe we need to be a bit more strict in that checking, if `other` is of a non-builtin type, stricter type checking should be in place, at least, it should be same class of sub-class of it

Reply:
ha, actually, it's a little weirder...
```Frame.worldXY() == Polyline([[0,0,0], [1,0,0], [0,1,0], [2,3,4])``` is true, and i was expecting that ```Polyline([[0,0,0], [1,0,0], [0,1,0], [2,3,4]) == Frame.worldXY()``` would return `False`, however there's a bug in  `Polyline.__eq__`.  so the following are both true..
```
Polyline([[0,0,0], [1,0,0], [0,1,0], [2,3,4]]) == Polyline([[0,0,0], [1,0,0], [0,1,0]])
Polyline([[0,0,0], [1,0,0], [0,1,0]]) == Polyline([[0,0,0], [1,0,0], [0,1,0], [2,3,4]])

Reply:
and i guess the same bug exists for polygon